artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/AbstractSequentialFile.java

```diff
@@ -301,21 +269,19 @@ protected ByteBuffer newBuffer(int size, int limit) {
    protected class LocalBufferObserver implements TimedBufferObserver {
```

Comment:
Is it worth extracting this to its own class file.

Reply:
It is not easy: it's `Local` to the owner class in the worst way: it use directly private files as `fileSize` that has a very tricky semantic around. I prefer to left it to stay where it is to reduce the amount of changes.